File: Data Structures & Algorithms/daily-temperatures/submission-0.rs
impl Solution {
    pub fn daily_temperatures(temperatures: Vec<i32>) -> Vec<i32> {
        let len = temperatures.len();
        let mut result = vec![0;len];
        let mut stack : Vec<usize> = Vec::new(); 
        for i in 0..len{
            if i == 0{
                stack.push(i);
            }
            else{
                loop{
                    let top = stack.last();
                    if top.is_some(){
                        let top_elem = *top.unwrap();
                        if temperatures[top_elem] < temperatures[i]{
                            result[top_elem] = i - top_elem;
                            stack.pop();
                        }
                        else{
                            stack.push(i);
                            break;
                        }
                    }
                    else{
                        stack.push(i);
                        break;
                    }
                }
            }
        }
        println!("{:?}",result);
        let mut result_final = vec![0;len];
        for i in 0..len{
            result_final[i] = result[i] as i32;
        }
        result_final
    }
}

File: Data Structures & Algorithms/count-number-of-islands/submission-0.java
class Solution {
    public int numIslands(char[][] grid) {
        int result = 0;
        for( int i = 0; i < grid.length; i++){
            for(int j = 0; j < grid[i].length; j++){
                if(grid[i][j] == '0'){
                    grid[i][j] = 'x';
                }
                else if(grid[i][j] == '1'){
                    grid = dfs(grid,i,j);
                    result += 1;
                }
            }
        }
        return result;
    }
    

    private char[][] dfs(char[][] grid, int i, int j){
        if(i < 0 || i >= grid.length){
            return grid;
        }
        if(j < 0 || j >= grid[i].length){
            return grid;
        }
        if(grid[i][j] != '1'){
            return grid;
        }
        else{
            grid[i][j] = 'x';
            grid = dfs(grid,i+1,j);
            grid = dfs(grid,i-1,j);
            grid = dfs(grid,i,j+1);
            grid = dfs(grid,i,j-1);
            return grid;
        }
    }
}
